R/navs-legacy.R

```diff
@@ -105,36 +105,223 @@ navset_hidden <- function(..., id = NULL, selected = NULL,
+    ...,
+    title = title,
+    id = id,
+    selected = selected,
+    sidebar = sidebar,
+    fillable = fillable,
+    gap = gap,
+    padding = padding,
+    header = header,
+    footer = footer,
+    fluid = fluid,
+    position = .navbar_options$position,
+    bg = .navbar_options$bg,
+    inverse = .navbar_options$inverse,
+    collapsible = .navbar_options$collapsible,
+    underline = .navbar_options$underline,
     # theme is only used to determine whether legacy style markup should be used
     # (and, at least at the moment, we don't need legacy markup for this exported function)
     theme = bs_theme()
   )
 }
 
+#' Create a set of navbar options
+#' 
+#' A `navbar_options()` object captures options specific to the appearance and
+#' behavior of the navbar, independent from the content displayed on the page.
+#' This helper should be used to create the list of options expected by
+#' `navbar_options` in [page_navbar()] and [navset_bar()].
+#' 
+#' ## Changelog
+#' 
+#' This function was introduced in \pkg{bslib} v0.9.0, replacing the `position`,
+#' `bg`, `inverse`, `collapsible` and `underline` arguments of [page_navbar()]
+#' and [navset_bar()]. Those arguments are deprecated with a warning and will be
+#' removed in a future version of \pkg{bslib}.
+#' 
+#' @examples
+#' navbar_options(position = "static-top", bg = "#2e9f7d", underline = FALSE)
+#' 
+#' @inheritParams shiny::navbarPage
+#' @param bg a CSS color to use for the navbar's background color.
+#' @param inverse Either `TRUE` for a light text color or `FALSE` for a dark
+#'   text color. If `"auto"` (the default), the best contrast to `bg` is chosen.
+#' @param underline Whether or not to add underline styling to page or navbar
+#'   links when active or focused.
+#' @param ... Additional arguments are ignored. `...` is included for future
+#'   expansion on `navbar_options()`.
+#' 
+#' @returns Returns a list of navbar options.
+#' 
+#' @export
+navbar_options <- function(
+  ...,
+  position = c("static-top", "fixed-top", "fixed-bottom"),
+  bg = NULL,
+  inverse = "auto",
+  collapsible = TRUE,
+  underline = TRUE
+) {
+  position2 <- rlang::arg_match(position)
+
+  opts <- list(
+    position = if (missing(position)) as_bslib_default(position[[1]]) else position2,
+    bg = if (missing(bg)) as_bslib_default(bg) else bg,
+    inverse = if (missing(inverse)) as_bslib_default(inverse) else inverse,
+    collapsible = if (missing(collapsible)) as_bslib_default(collapsible) else collapsible,
+    underline = if (missing(underline)) as_bslib_default(underline) else underline
+  )
+  
+  structure(dropNulls(opts), class = c("bslib_navbar_options", "list"))
+}
+
+as_bslib_default <- function(x) {
+  if (is.null(x)) return(x)
+  attr(x, "bslib_default") <- TRUE
+  attr(x, "waldo_opts") <- list(ignore_attr = TRUE)
+  x
+}
+
+is_bslib_default <- function(x) {
+  isTRUE(attr(x, "bslib_default"))
+}
+
+navbar_options_resolve_deprecated <- function(
+  options_user = NULL,
```

Comment:
If we make it more clear this is a `list()`, you could get rid of 271

```suggestion
  options_user = navbar_options(),
```